larq_zoo/densenet.py

```diff
@@ -0,0 +1,181 @@
+from zookeeper import registry, HParams
+from tensorflow import keras
+from larq_zoo import utils
+import larq as lq
+
+
+def densely_connected_block(x, args):
+    y = keras.layers.BatchNormalization(momentum=0.9, epsilon=1e-5)(x)
+    y = lq.layers.QuantConv2D(
+        filters=args.growth_rate,
+        kernel_size=3,
+        input_quantizer=args.quantizer,
+        kernel_quantizer=args.quantizer,
+        kernel_initializer="glorot_normal",
+        kernel_constraint=args.constraint,
+        padding="same",
+        use_bias=False,
+    )(y)
+    return keras.layers.concatenate([x, y])
+
+
+@registry.register_model
+def densenet(args, dataset, input_tensor=None, include_top=True):
+    input = utils.get_input_layer(dataset.input_shape, input_tensor)
+
+    if dataset.input_shape[0] and dataset.input_shape[0] < 50:
```

Comment:
Maybe it would be good to have a comment somewhere on how the imageshape of the dataset should look like for all the models we have. Especially if there are different possible image sizes as here.

Reply:
It is mentioned in the docstrings. This line is mostly in order to allow training with CIFAR10